lms/djangoapps/certificates/queue.py

```diff
@@ -237,8 +253,24 @@ def add_cert(self, student, course_id, course=None, forced_grade=None, template_
             status.unverified,
         ]
 
-        cert_status = certificate_status_for_student(student, course_id)['status']
+        cert_status_dict = certificate_status_for_student(student, course_id)
+        cert_status = cert_status_dict['status']
         cert = None
+        if cert_status_dict.get('download_url'):
+            LOGGER.warning(
```

Comment:
would be nice to have it as common function because you are using the same piece of code in 2 files.

Reply:
@adeelehsan I have refactored both queue and test, kindly take another look.